include/RAJA/util/PluginContext.hpp

```diff
@@ -21,9 +23,13 @@ class KokkosPluginLoader;
 struct PluginContext
 {
 public:
-  PluginContext(const Platform p) : platform(p) {}
+  PluginContext(const Platform p, const std::string* name = nullptr)
```

Comment:
Is this taking the name by pointer to make PluginContext copyable? Does it make more sense to take the string by reference, or string_view once we have it? We could use a static null string as the default.